### PR TITLE
COMPAT: Upstream compat

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref https://github.com/dask/dask/issues/4961

Doesn't quite fix them all. A few text failures related to compression.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
I have a couple things to follow up on

- https://github.com/pandas-dev/pandas/issues/27190
- ...

**Comment by jrbourbeau** [comment]
I merged `master` into this branch as the `fsspec` work in #5064 should fix some of the test failures on Travis. Hope that's okay

**Comment by TomAugspurger** [comment]
Thanks @jrbourbeau. I think this is getting close (and we're on a bit of a deadline, since the pandas release happened yesterday).

The remaining failures may just be compression related ones, like https://travis-ci.org/dask/dask/jobs/560820240#L2014. I can't reproduce those locally yet.

**Comment by TomAugspurger** [comment]
At least some CI builds have started using pandas 0.25, so I'm probably going to merge this when CI passes (anyone have time for a quick glance over the changes? @jrbourbeau?).

I think that the upstream job will still be failing on the compression changes. I'll follow up with another PR finishing those off. But at least the CI will be passing again (upstream-dev is an allowed failure).

File: continuous_integration/travis/install.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ if [[ ${UPSTREAM_DEV} ]]; then
         locket \
         git+https://github.com/pydata/sparse \
         git+https://github.com/dask/s3fs \
-        git+https://github.com/intake/fsspec \
+        git+https://github.com/intake/filesystem_spec \
```
Comment (martindurant) [inline continuous_integration/travis/install.sh]: Is this actually getting built? I would have noticed, I think

Comment (TomAugspurger) [reply continuous_integration/travis/install.sh]: It's built nightly.

On PRs, you can trigger a build by including `test-upstream` in the message.

Comment (martindurant) [reply continuous_integration/travis/install.sh]: good to know.

File: dask/dataframe/multi.py [hunk 2]
```diff
@@ -927,11 +921,7 @@ def stack_partitions(dfs, divisions, join="outer"):
     for df in dfs:
         # dtypes of all dfs need to be coherent
         # refer to https://github.com/dask/dask/issues/4685
-        if is_dataframe_like(df):
```
Comment (jrbourbeau) [inline dask/dataframe/multi.py]: Why are we removing this `is_dataframe_like` check?

File: dask/dataframe/tests/test_arithmetics_reduction.py [hunk 3]
```diff
@@ -992,7 +1011,8 @@ def check_raises(d, p, func):
         check_raises(dds, pds, "std")
         check_raises(dds, pds, "var")
         check_raises(dds, pds, "sem")
-        check_raises(dds, pds, "mean")
+        if not (PANDAS_GT_0250 and is_datetime64_ns_dtype(pds.dtype)):
+            check_raises(dds, pds, "mean")
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_arithmetics_reduction.py]: Will be useful for knowing when to remove the `PANDAS_GT_0250` check 

```suggestion
            # pandas 0.25 added DatetimeIndex.mean. We need to follow
            check_raises(dds, pds, "mean")
```

File: dask/dataframe/multi.py (hunk 2 above)
Comment (TomAugspurger) [reply dask/dataframe/multi.py]: Previously we did `df.columns.intersection(DataFrame)` which was just plain wrong. That always ended up in an empty `shared` so the `.astype` never occurred.

When I changed things to `df.columns.interesction(meta.columns)` to match the intent of the original code, I got failures with know & unknown categoricals.

**Comment by TomAugspurger** [comment]
Thanks. Merging on green.

**Comment by TomAugspurger** [comment]
Just the partd failure in https://ci.appveyor.com/project/daskdev/dask/builds/26105683, so as green as it gets right now.

**Pull request merged**